app/frontend/src/components/ches/UploadContexts.vue

```diff
@@ -0,0 +1,385 @@
+      @dragover.prevent="dragover = true"
+      @dragenter.prevent="dragover = true"
+      @dragleave.prevent="dragover = false"
+      :class="['drop-zone', { 'grey lighten-2': dragover }]"
+    >
+      <v-container>
+        <v-row
+          class="d-flex flex-column"
+          dense
+          align="center"
+          justify="center"
+        >
+          <p class="pt-5 mb-1">
+            Drag &amp; Drop your Excel file here, or click to Browse.
+          </p>
+          <p class="">
+            A sample Excel file containing contexts is available to download on the <router-link :to="{ name: 'MergeAbout'}" >About</router-link> tab.
+          </p>
+        </v-row>
+      </v-container>
+
+      <v-file-input
+        class="d-none"
+        @change="fileInputChange($event)"
+        ref="fileInput"
+        label="File input"
+      ></v-file-input>
+    </div>
+  </div>
+</template>
+
+<script>
+/**
+ * This component handles the UI for adding an Excel file of contexts for the mail merge.
+ * The Excel file is parsed and the contexts within it
+ * are added to the vuex store ('mergeForm.contexts' property)
+ * The contexts are also displayed to the user in a table
+ */
+
+import XLSX from 'xlsx';
+import moment from 'moment';
+import { mapActions } from 'vuex';
+import { mapFields } from 'vuex-map-fields';
+
+import { Contexts, Formats } from '@/utils/constants';
+import * as Utils from '@/utils/utils';
+
+export default {
+  name: 'UploadContexts',
+
+  data() {
+    return {
+      dragover: false,
+    };
+  },
+
+  computed: {
+    ...mapFields('ches', [
+      'mergeForm.excelParsed',
+      'mergeForm.excel',
+      'mergeForm.contexts',
+      'mergeForm.contextsType',
+      'mergeForm.contextVariables',
+    ]),
+  },
+
+  methods: {
+    ...mapActions('alert', ['showAlert', 'clearAlert']),
+
+    addExcelFile(e) {
+      this.dragover = false;
+
+      const file = e.dataTransfer.file;
+      // validate Excel file
+      if (!this.validateFile(file)) {
+        // show error alert
+        this.alert = {
+          type: 'warning',
+          text: 'Context file size cannot excede 20MB',
+        };
+        this.showAlert(this.alert);
+      }
+      // else attachments are valid
+      else {
+        // parse file
+        this.parseFile(file, this.addContextsToStore);
+        // make file data avalable in parent component
+        this.$emit('fileUploaded', file);
+        // show table of contexts
+        this.excelParsed = true;
+
+        this.clearAlert();
+      }
+    },
+
+    parseFile(file, cb) {
+      // this will suppress a console warning about moment deprecating a default fallback on non ISO/RFC2822 date formats
+      // we will just force it to use the new Date constructor.
+      moment.createFromInputFallback = function (config) {
+        config._d = new Date(config._i);
+      };
+
+      const reader = new FileReader();
+      const rABS = !!reader.readAsBinaryString;
```

Comment:
This is one of the few cases where I'd recommend just in-lining the `reader.readAsBinaryString` where it's needed instead of using an rABS variable as it would improve readability and clarity on what it's representing.
Also note that at the time of evaluation and assignment, it is possible that the reader has not actually begun reading the file stream, so this may actually be assigning something you might not be expecting due to order of operations.